packages/streaming-image-volume-loader/src/BaseStreamingImageVolume.ts

```diff
@@ -204,319 +220,357 @@ export default class BaseStreamingImageVolume extends ImageVolume {
 
-    // SharedArrayBuffer
-    const arrayBuffer = scalarData.buffer;
-    const numFrames = imageIds.length;
+    if (complete && this.framesProcessed === this.totalNumFrames) {
+      this.loadStatus.loaded = true;
+      this.loadStatus.loading = false;
+    }
 
-    // Length of one frame in voxels
-    const length = scalarData.length / numFrames;
-    // Length of one frame in bytes
-    const lengthInBytes = arrayBuffer.byteLength / numFrames;
+    this.callLoadStatusCallback({
+      success: true,
+      imageIdIndex,
+      imageId,
+      framesLoaded: this.framesLoaded,
+      framesProcessed: this.framesProcessed,
+      framesUpdated: this.framesUpdated,
+      numFrames,
+      totalNumFrames,
+      complete,
+      imageQualityStatus,
+    });
+    if (this.loadStatus.loaded) {
+      this.loadStatus.callbacks = [];
+    }
+  }
 
-    let type;
+  public successCallback(imageId: string, image) {
+    const imageIdIndex = this.getImageIdIndex(imageId);
+    const options = this.getLoaderImageOptions(imageId);
+    const scalarData = this._getScalarDataByImageIdIndex(imageIdIndex);
+    handleArrayBufferLoad(scalarData, image, options);
 
-    if (scalarData instanceof Uint8Array) {
-      type = 'Uint8Array';
-    } else if (scalarData instanceof Float32Array) {
-      type = 'Float32Array';
-    } else if (scalarData instanceof Uint16Array) {
-      type = 'Uint16Array';
-    } else if (scalarData instanceof Int16Array) {
-      type = 'Int16Array';
-    } else {
-      throw new Error('Unsupported array type');
-    }
+    const { scalingParameters } = image.preScale || {};
+    const { imageQualityStatus } = image;
+    const frameIndex = this._imageIdIndexToFrameIndex(imageIdIndex);
 
-    const totalNumFrames = this.imageIds.length;
-    const autoRenderOnLoad = true;
-    const autoRenderPercentage = 2;
+    // Check if there is a cached image for the same imageURI (different
+    // data loader scheme)
+    const cachedImage = cache.getCachedImageBasedOnImageURI(imageId);
 
-    let reRenderFraction;
-    let reRenderTarget;
+    // Check if the image was already loaded by another volume and we are here
+    // since we got the imageLoadObject from the cache from the other already loaded
+    // volume
+    const cachedVolume = cache.getVolumeContainingImageId(imageId);
 
-    if (autoRenderOnLoad) {
-      reRenderFraction = totalNumFrames * (autoRenderPercentage / 100);
-      reRenderTarget = reRenderFraction;
+    // check if the load was cancelled while we were waiting for the image
+    // if so we don't want to do anything
+    if (this.loadStatus.cancelled) {
+      console.warn(
+        'volume load cancelled, returning for imageIdIndex: ',
+        imageIdIndex
+      );
+      return;
     }
 
-    function callLoadStatusCallback(evt) {
-      // TODO: probably don't want this here
+    // if it is not a cached image or volume
+    if (!cachedImage && !(cachedVolume && cachedVolume.volume !== this)) {
```

Comment:
@sedghi - this line was wrong previously and for a cached image response which is still waiting to receive data, it
won't got through handle from cache code, which mucks up the display.  I believe the stack/volume caches are a bit more efficient and allow the stack cache to get started but not completed before the volume cache checks.

Reply:
I think i know what was happening, it is reusing the image from stack viewport and that has an issue as you saw